nvtabular/dask/io.py

```diff
@@ -184,7 +195,19 @@ def __init__(
                     "Using very large partitions sizes for Dask. "
                     "Memory-related errors are likely."
                 )
-            part_size = int(cuda.current_context().get_memory_info()[1] * part_mem_fraction)
+
+            try:
+                part_size = int(cuda.current_context().get_memory_info()[1] * part_mem_fraction)
+            except NotImplementedError:
+                # TODO: Remove this block when rmm approach is fully supported
+                import pynvml
+
+                pynvml.nvmlInit()
+                part_size = int(
+                    pynvml.nvmlDeviceGetMemoryInfo(pynvml.nvmlDeviceGetHandleByIndex(0)).total
+                    * part_mem_fraction
+                )
+                pynvml.nvmlShutdown()
```

Comment:
Rather than have this block of code here and a similar block in the examples script - can we pull out to a common function?

Also - is there a reason to even try the cuda.current_context() code here, rather than just always getting memory info from pynvml?

Reply:
Agreed - I haven't been tracking how likely the numba/rmm issue is to be fixed, so I was assuming this "workaround" may be removed soon.  However, we might as well make the pynvml approach a bit more standardized if it is here to stay :)

Reply:
The current proposal seems to indicate that you can enable a tracking resource adaptor to get a sense of how much memory is free in RMM: https://github.com/rapidsai/rmm/issues/415 .

I think right now we should standardize on the pynvml approach- but it doesn't need to be part of this PR. I think many of the other calls to rmm.get_info are going to go away in the next week or so as we finish the dask migration

Reply:
Done in #107